09:39:59          c:\tf_jenkins\home\workspace\tensorflow-pr-win-cmake-py\tensorflow\core\framework\bfloat16.cc(23): error C2065: '__BYTE_ORDER__': undeclared identifier [C:\tf_jenkins\home\workspace\tensorflow-pr-win-cmake-py\cmake_build\tf_core_framework.vcxproj]
09:39:59          c:\tf_jenkins\home\workspace\tensorflow-pr-win-cmake-py\tensorflow\core\framework\bfloat16.cc(23): error C2065: '__ORDER_BIG_ENDIAN__': undeclared identifier [C:\tf_jenkins\home\workspace\tensorflow-pr-win-cmake-py\cmake_build\tf_core_framework.vcxproj]
09:39:59          c:\tf_jenkins\home\workspace\tensorflow-pr-win-cmake-py\tensorflow\core\framework\bfloat16.cc(37): error C2065: '__BYTE_ORDER__': undeclared identifier [C:\tf_jenkins\home\workspace\tensorflow-pr-win-cmake-py\cmake_build\tf_core_framework.vcxproj]
09:39:59          c:\tf_jenkins\home\workspace\tensorflow-pr-win-cmake-py\tensorflow\core\framework\bfloat16.cc(37): error C2065: '__ORDER_BIG_ENDIAN__': undeclared identifier [C:\tf_jenkins\home\workspace\tensorflow-pr-win-cmake-py\cmake_build\tf_core_framework.vcxproj]

--- a/tensorflow/core/BUILD
+++ b/tensorflow/core/BUILD
@@ -240,7 +240,6 @@ cc_library(
         "lib/strings/strcat.h",
         "lib/strings/stringprintf.h",
         "platform/cpu_info.h",
-        "platform/windows/cpu_info.h",
         "platform/dynamic_annotations.h",
         "platform/env.h",
         "platform/file_system.h",
@@ -260,6 +259,7 @@ cc_library(
         "platform/subprocess.h",
         "platform/thread_annotations.h",
         "platform/types.h",
+        "platform/windows/cpu_info.h",
     ],
     visibility = ["//visibility:public"],
     deps = [

--- a/tensorflow/core/BUILD
+++ b/tensorflow/core/BUILD
@@ -240,6 +240,7 @@ cc_library(
         "lib/strings/strcat.h",
         "lib/strings/stringprintf.h",
         "platform/cpu_info.h",
+        "platform/windows/cpu_info.h",
         "platform/dynamic_annotations.h",
         "platform/env.h",
         "platform/file_system.h",

--- a/tensorflow/core/framework/bfloat16.cc
+++ b/tensorflow/core/framework/bfloat16.cc
@@ -20,31 +20,31 @@ namespace tensorflow {
 void FloatToBFloat16(const float* src, bfloat16* dst, int64 size) {
   const uint16_t* p = reinterpret_cast<const uint16_t*>(src);
   uint16_t* q = reinterpret_cast<uint16_t*>(dst);
-  if (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__) {  
-    for (; size; p += 2, q++, size--) {  
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+    for (; size != 0; p += 2, q++, size--) {  
       *q = p[0];  
     }  
-  } else {  
-    for (; size; p += 2, q++, size--) {  
+#else
+    for (; size != 0; p += 2, q++, size--) {  
      *q = p[1];  
     }  
-  }
+#endif
 }
 
 void BFloat16ToFloat(const bfloat16* src, float* dst, int64 size) {
   const uint16_t* p = reinterpret_cast<const uint16_t*>(src);
   uint16_t* q = reinterpret_cast<uint16_t*>(dst);
-  if (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__) {  
-    for (; size; p++, q += 2, size--) {  
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+    for (; size != 0; p++, q += 2, size--) {  
       q[0] = *p;  
       q[1] = 0;  
-    }  
-  } else {  
-    for (; size; p++, q += 2, size--) {  
+    }
+#else  
+    for (; size != 0; p++, q += 2, size--) {  
       q[0] = 0;  
       q[1] = *p;  
     } 
-  }
+#endif
 }
 
 }  // end namespace tensorflow

--- a/tensorflow/core/framework/bfloat16.h
+++ b/tensorflow/core/framework/bfloat16.h
@@ -19,6 +19,10 @@ limitations under the License.
 #include "tensorflow/core/framework/numeric_types.h"
 #include "tensorflow/core/platform/types.h"
 
+#if defined(PLATFORM_WINDOWS)  
+#include "tensorflow/core/platform/windows/cpu_info.h"  
+#endif  
+
 // Compact 16-bit encoding of floating point numbers. This representation uses
 // 1 bit for the sign, 8 bits for the exponent and 7 bits for the mantissa.  It
 // is assumed that floats are in IEEE 754 format so the representation is just

--- a/tensorflow/core/kernels/cast_op.h
+++ b/tensorflow/core/kernels/cast_op.h
@@ -102,7 +102,7 @@ struct scalar_cast_op<::tensorflow::bfloat16, float> {
     p[0] = a.value;  
     p[1] = 0;  
 #else  
-    static_assert(::tensorflow::port::kLittleEndian, "");  
+    static_assert(::tensorflow::port::kLittleEndian, "Not a little endian system!");  
     p[0] = 0;
     p[1] = a.value;
 #endif
@@ -125,7 +125,7 @@ struct scalar_cast_op<float, ::tensorflow::bfloat16> {
     const uint16_t* p = reinterpret_cast<const uint16_t*>(&a);  
     return ::tensorflow::bfloat16(p[0]);  
 #else 
-    static_assert(::tensorflow::port::kLittleEndian, "");
+    static_assert(::tensorflow::port::kLittleEndian, "Not a little endian system!");
     const uint16_t* p = reinterpret_cast<const uint16_t*>(&a);
     return ::tensorflow::bfloat16(p[1]);
 #endif 

--- a/tensorflow/core/kernels/sparse_matmul_op.cc
+++ b/tensorflow/core/kernels/sparse_matmul_op.cc
@@ -258,11 +258,11 @@ static const int kNumOperands = (sizeof(Packet) / sizeof(float));
 ALWAYS_INLINE float ConvertBfloat16ToFloat(const bfloat16* src) {
   float out = 0;
   auto tmp = reinterpret_cast<bfloat16*>(&out);
-  if (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__) {
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
     tmp[0] = *src;
-  } else {
+#else
     tmp[1] = *src;
-  }
+#endif
   return out;
 }
 

--- a/tensorflow/core/kernels/sparse_matmul_op.h
+++ b/tensorflow/core/kernels/sparse_matmul_op.h
@@ -20,6 +20,7 @@ limitations under the License.
 #include "tensorflow/core/platform/types.h"
 
 #if defined(PLATFORM_WINDOWS)
+#include "tensorflow/core/platform/windows/cpu_info.h"
 #include "tensorflow/core/platform/windows/intrinsics_port.h"
 #endif
 
@@ -31,11 +32,11 @@ namespace internal {
 template <typename Packet>
 EIGEN_DEVICE_FUNC inline Packet pexpand_bf16_l(const Packet& from) {
   tensorflow::uint32 tmp;  
-  if ( __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__ ) {  
-       tmp = (reinterpret_cast<const tensorflow::uint32&>(from) ) & 0xffff0000;  
-  } else {  
-       tmp = (reinterpret_cast<const tensorflow::uint32&>(from) << 16) & 0xffff0000;  
-  }  
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+    tmp = (reinterpret_cast<const tensorflow::uint32&>(from) ) & 0xffff0000;  
+#else    
+    tmp = (reinterpret_cast<const tensorflow::uint32&>(from) << 16) & 0xffff0000;  
+#endif
   return reinterpret_cast<const float&>(tmp);
 }
 
@@ -44,11 +45,11 @@ EIGEN_DEVICE_FUNC inline Packet pexpand_bf16_l(const Packet& from) {
 template <typename Packet>
 EIGEN_DEVICE_FUNC inline Packet pexpand_bf16_u(const Packet& from) {
   tensorflow::uint32 tmp;  
-  if ( __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__ ) {  
-       tmp = (reinterpret_cast<const tensorflow::uint32&>(from) << 16 ) & 0xffff0000;  
-  } else {  
-       tmp = (reinterpret_cast<const tensorflow::uint32&>(from)) & 0xffff0000;  
-  } 
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+    tmp = (reinterpret_cast<const tensorflow::uint32&>(from) << 16 ) & 0xffff0000;  
+#else
+    tmp = (reinterpret_cast<const tensorflow::uint32&>(from)) & 0xffff0000;  
+#endif 
   return reinterpret_cast<const float&>(tmp);
 }
 

--- a/tensorflow/core/kernels/sparse_matmul_op_test.cc
+++ b/tensorflow/core/kernels/sparse_matmul_op_test.cc
@@ -207,13 +207,13 @@ class SparseMatmulOpTest : public ::testing::Test {
       uint16_t* data3_p = reinterpret_cast<uint16_t*>(&data3[i]);
       uint16_t* data3_bfloat16_p =
           reinterpret_cast<uint16_t*>(data3_bfloat16) + i;
-      if ( __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__ ) {  
-           data3_p[1] = 0;  
-           data3_bfloat16_p[0] = data3_p[0];  
-      } else {  
-           data3_p[0] = 0;  
-           data3_bfloat16_p[0] = data3_p[1];  
-      }  
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+            data3_p[1] = 0;  
+            data3_bfloat16_p[0] = data3_p[0];  
+#else
+            data3_p[0] = 0;  
+            data3_bfloat16_p[0] = data3_p[1];  
+#endif  
     }
   }
 

